### DISCUSSION TITLE
Welcome to Coqui-AI TTS Discussions!

### DISCUSSION THREAD

**Discussion by erogol** [coqui-ai/TTS#8] [Announcements]
# 🐸 COQUI Discussions
### 👋 Welcome!
 We’re using Discussions as a place to connect with other members of our community. We hope that you ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  Feel free to comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by erogol** [comment]
ignore (move the thread up again)

**Comment by hamxa678** [comment]
Can i have hindi accent voice cloning here?

**Comment by N0vato** [comment]
@erogol 
greetings,
why isn't there a tutorial to run google colab in detail ?
can you make a video tutorial and share it please ?
(https://colab.research.google.com/drive/1NC4eQJFvVEqD8L4Rd8CVK25_Z-ypaBHD?usp=sharing)

**Reply by thorstenMueller** [reply]
What would you expect from a video about this notebook? Just executing it, the way it is or modifying it in some way (other dataset, other model/vocoder, ...).

**Reply by N0vato** [reply]
@thorstenMueller 

Days ago I was on your youtube channel looking for a tutorial since I saw, that you are the only one talking about coqui tts I started watching your videos, but I did not find a tutorial running google notebook, and I wonder, what is the reason why you have not made the tutorial or there are more people talking about coqui tts ?

**Comment by N0vato** [comment]
Good morning,

For now I just want to run google colab. You see, I'm interested because I
saw a voice cloning project that a guy did his college thesis on. I was
very interested to see how amazing it was to clone the voice and be able to
make scripts to create content or big projects from a few seconds of my
voice supplied.

The problem with that cloning project is that it didn't work very well and
the guy can't maintain it because he is always working. What the guy did
was to recommend your clone project, which is better structured and better
maintained.

The thing is that I was very surprised with what artificial intelligence
can do and I decided to learn about it.

Now I'm learning English and looking for articles about artificial
intelligence, while I study other interesting projects.

I want to know how to run this colab to see what it is capable of and maybe
be able to do things I couldn't do with the other github project.

That's why I ask you to please make a tutorial to test the colab and study
it deeper. I only ask that until I can handle English much better and learn
how to solve the problems I will encounter when I try to run another colab.

**Comment by N0vato** [comment]
I can't install almost anything on my laptop because it is old and I can
only run the colab.

**Reply by p0p4k** [reply]
This [video](https://www.youtube.com/watch?v=Ilj1E42p9aI) might be helpful.

**Reply by N0vato** [reply]
@p0p4k

It's great everything, but can you make a video with the notebook that is in this link ?

https://colab.research.google.com/drive/1NC4eQJFvVEqD8L4Rd8CVK25_Z-ypaBHD?usp=sharing

**Reply by p0p4k** [reply]
@Fulanito0 the notebook is using outdated repo. Please read the [documentation](https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html) and try to mimic that in a new colab notebook. It will help you understand the project better.
Also, for voice cloning [this](https://github.com/Edresson/YourTTS) is a 🐸-TTS model  with a working colab notebook.

**Comment by xianxiaoyuehub** [comment]
What would be appropriate for the loss of loss_align and total loss about the fast_pitch model trained on vctk dataset

**Comment by BasToTheMax** [comment]
What is the fastest way to train my own TTS model?

**Comment by TaterTotX** [comment]
I've been trying to download TTS on my M1 Mac for three days and finally succeeded. However, I keep getting an error message saying '[!] Model file not found in the output path.' Could someone please help me? Thank you very much.





from TTS.api import TTS

# List available 🐸TTS models and choose the first one
model_name = TTS.list_models()[0]
# Init TTS
tts = TTS(model_name)
# Run TTS
# ❗ Since this model is multi-speaker and multi-lingual, we must set the target speaker and the language
# Text to speech with a numpy output
wav = tts.tts("This is a test! This is also a test!!", speaker=tts.speakers[0], language=tts.languages[0])
# Text to speech to a file
tts.tts_to_file(text="Hello world!", speaker=tts.speakers[0], language=tts.languages[0], file_path="output.wav")

raise ValueError(" [!] Model file not found in the output path")
ValueError:  [!] Model file not found in the output path
 > tts_models/multilingual/multi-dataset/your_tts is already downloaded.

**Comment by Ruebezahl11** [comment]
Hi, I am trying to use coqui to speak german texts. There is a problem with numbers. Suppose the phrase is "findet am 28. August statt". tts will split the input into what it thinks are sentences, and detect a break sentence break at 28. Is there a way to mask the period? Replacing t with 28ten or something like that is not an option, because I would need to understand the entire phrase in order to know the right genus

**Reply by p0p4k** [reply]
see the code and comment out the pybsd  segmenting stuff.

**Comment by Ruebezahl11** [comment]
Hi, thanks for pointing at the segmenting stuff. It turns out that the script is hardcoded for english; I justhadto change it to german.
Unfortunately the number with dot is still not pronounced as an ordinal number
Aside question: can tts speak phonetic (IPA) codes?

**Reply by p0p4k** [reply]
If the model was trained on IPA, it can speak that; else not.

**Comment by Ruebezahl11** [comment]
@thorstenMueller
Hi, can your voices speak phonetic?

**Reply by thorstenMueller** [reply]
Sorry for late response. You mean input phonemes instead of actual text string?

**Reply by Ruebezahl11** [reply]
Hi Thorsten, something like that. In the meantime I have seen your video about adding to espeak dicts, but I originally had the idea to create a list of proper names and their IPA pronounciations.
BTW: I am still trying to set up coqui tts with your voices. The vits works fine, but in tacotron all umlaut and ß are missing. Do you have an idea what could be wrong?
One more question: is there a processing step that could be made to read "Am 3. Oktober" or "der 4. Juli" correctly

**Reply by thorstenMueller** [reply]
Hi @Ruebezahl11 , are you using Windows? I encountered problems with german Umlauts on Windows using eSpeak-ng. These were gone after using eSpeak, the classical without -ng. Could you try this?
I have written some tips on this page under my Windows tutorial video.
https://www.thorsten-voice.de/einfach-loslegen/

And about the date problem, the model will consider a dot as end of sentence, but you can try this as text cleaner before running TTS. It will create a "am dritten Oktober".
https://github.com/repodiac/german_transliterate

**Comment by JahangirMoro** [comment]
I want to train my tts on new language( urdu ) can you plese help how can i do it with couqi tts ...

**Reply by Mushtaq25** [reply]
Create the dataset like ljspeech format (given on the docs) and use phonemes= True, set phonemes_language="", inside config.json, Rest mentioned in the docs( https://docs.coqui.ai/en/latest/tutorial_for_nervous_beginners.html)

**Comment by Mushtaq25** [comment]
Hi I am little bit confused about the train.py file( https://docs.coqui.ai/en/latest/tutorial_for_nervous_beginners.html ) while training, it is also generating audio file i can check in the wandb, so my question is it is not neccesary to train the vocoder.py, or it is must to train both glow-tss and vocoder for inference.

**Comment by Alekh-sinha** [comment]
Can it generate speech with accent? I am trying to generate text in SRK voice (Indian film star). It is picking up the voice but not the accent. Can anybody help me in this?

**Comment by Tpyter** [comment]
Traceback (most recent call last):
  File "D:\COTTS\myenvtts\lib\site-packages\urllib3\response.py", line 748, in _error_catcher
    yield
  File "D:\COTTS\myenvtts\lib\site-packages\urllib3\response.py", line 873, in _raw_read
    data = self._fp_read(amt, read1=read1) if not fp_closed else b""
  File "D:\COTTS\myenvtts\lib\site-packages\urllib3\response.py", line 856, in _fp_read
    return self._fp.read(amt) if amt is not None else self._fp.read()
  File "D:\python\lib\http\client.py", line 463, in read
    n = self.readinto(b)
  File "D:\python\lib\http\client.py", line 507, in readinto
    n = self.fp.readinto(b)
  File "D:\python\lib\socket.py", line 704, in readinto
    return self._sock.recv_into(b)
  File "D:\python\lib\ssl.py", line 1241, in recv_into
    return self.read(nbytes, buffer)
  File "D:\python\lib\ssl.py", line 1099, in read
    return self._sslobj.read(len, buffer)
socket.timeout: The read operation timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\COTTS\myenvtts\lib\site-packages\requests\models.py", line 820, in generate
    yield from self.raw.stream(chunk_size, decode_content=True)
  File "D:\COTTS\myenvtts\lib\site-packages\urllib3\response.py", line 1060, in stream
    data = self.read(amt=amt, decode_content=decode_content)
  File "D:\COTTS\myenvtts\lib\site-packages\urllib3\response.py", line 949, in read
    data = self._raw_read(amt)
  File "D:\COTTS\myenvtts\lib\site-packages\urllib3\response.py", line 902, in _raw_read
    self._fp.close()
  File "D:\python\lib\contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "D:\COTTS\myenvtts\lib\site-packages\urllib3\response.py", line 753, in _error_catcher
    raise ReadTimeoutError(self._pool, None, "Read timed out.") from e  # type: ignore[arg-type]
urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='user-images.githubusercontent.com', port=443): Read timed out.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\COTTS\TTS\recipes\ljspeech\glow_tts\train_glowtts.py", line 79, in <module>
    trainer = Trainer(
  File "D:\COTTS\myenvtts\lib\site-packages\trainer\trainer.py", line 583, in __init__
    ping_training_run()
  File "D:\COTTS\myenvtts\lib\site-packages\trainer\analytics.py", line 12, in ping_training_run
    _ = requests.get(URL, timeout=5)
  File "D:\COTTS\myenvtts\lib\site-packages\requests\api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "D:\COTTS\myenvtts\lib\site-packages\requests\api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "D:\COTTS\myenvtts\lib\site-packages\requests\sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "D:\COTTS\myenvtts\lib\site-packages\requests\sessions.py", line 724, in send
    history = [resp for resp in gen]
  File "D:\COTTS\myenvtts\lib\site-packages\requests\sessions.py", line 724, in <listcomp>
    history = [resp for resp in gen]
  File "D:\COTTS\myenvtts\lib\site-packages\requests\sessions.py", line 265, in resolve_redirects
    resp = self.send(
  File "D:\COTTS\myenvtts\lib\site-packages\requests\sessions.py", line 746, in send
    r.content
  File "D:\COTTS\myenvtts\lib\site-packages\requests\models.py", line 902, in content
    self._content = b"".join(self.iter_content(CONTENT_CHUNK_SIZE)) or b""
  File "D:\COTTS\myenvtts\lib\site-packages\requests\models.py", line 826, in generate
    raise ConnectionError(e)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='user-images.githubusercontent.com', port=443): Read timed out.
How can we solve this kind of problem?

**Reply by eginhard** [reply]
This is probably just a temporary connection error. However, feel free to use [our fork](https://github.com/idiap/coqui-ai-TTS) (available via `pip install coqui-tts`), which removed the telemetry in the Trainer and should avoid this in the future.

**Comment by Tpyter** [comment]
Hello, I would like to run a project on Windows. Could you provide me with specific training instructions? Also, I have a GPU and I hope to run the project on the GPU on Windows.

**Comment by Tpyter** [comment]
D:\AAAAA\myenv\Scripts\python.exe D:/AAAAA/coqui-ai-TTS/recipes/ljspeech/glow_tts/train_glowtts.py
 > Training Environment:
 | > Backend: Torch
 | > Mixed precision: True
 | > Precision: fp16
 | > Num. of CPUs: 12
 | > Num. of Torch Threads: 1
 | > Torch seed: 54321
 | > Torch CUDNN: True
 | > Torch CUDNN deterministic: False
 | > Torch CUDNN benchmark: False
 | > Torch TF32 MatMul: False
 > Start Tensorboard: tensorboard --logdir=D:\AAAAA\coqui-ai-TTS\recipes\ljspeech\glow_tts\run-August-16-2024_06+12PM-204588f7

 > Model has 28610257 parameters

 > EPOCH: 0/1000
 --> D:\AAAAA\coqui-ai-TTS\recipes\ljspeech\glow_tts\run-August-16-2024_06+12PM-204588f7
  0%|          | 2/12969 [00:00<45:33,  4.74it/s]  ɪnstɛd əv weɪtɪŋ ðɛɹ, ɔzwɔld əpɛɹəntli wɛnt æz fɑɹ əweɪ æz hi kʊd ænd bɔɹdɪd ðə fɚst oʊk klɪf bʌs wɪt͡ʃ keɪm əlɔŋ
Character '͡' not found in the vocabulary. Discarding it.
 16%|█▌        | 2054/12969 [01:53<11:13, 16.21it/s]ɪntu ðə “kɹeɪtɚ” dʌɡ aʊt ɪn ðə mɪdəl, pɔɹ ðə spʌnd͡ʒ, wɔɹm wɔtɚ, ðə məlæsɪz, ænd soʊdə dɪzɑlvd ɪn hɑt wɔtɚ.
Character '“' not found in the vocabulary. Discarding it.
ɪntu ðə “kɹeɪtɚ” dʌɡ aʊt ɪn ðə mɪdəl, pɔɹ ðə spʌnd͡ʒ, wɔɹm wɔtɚ, ðə məlæsɪz, ænd soʊdə dɪzɑlvd ɪn hɑt wɔtɚ.
Character '”' not found in the vocabulary. Discarding it.
100%|██████████| 12969/12969 [08:57<00:00, 24.12it/s]

 > TRAINING (2024-08-16 18:21:51) 
 > Training Environment:
 | > Backend: Torch
 | > Mixed precision: True
 | > Precision: fp16
 | > Num. of CPUs: 12
 | > Num. of Torch Threads: 1
 | > Torch seed: 54321
 | > Torch CUDNN: True
 | > Torch CUDNN deterministic: False
 | > Torch CUDNN benchmark: False
 | > Torch TF32 MatMul: False
 > Start Tensorboard: tensorboard --logdir=D:\AAAAA\coqui-ai-TTS\recipes\ljspeech\glow_tts\run-August-16-2024_06+21PM-204588f7

 > Model has 28610257 parameters

 > EPOCH: 0/1000
 --> D:\AAAAA\coqui-ai-TTS\recipes\ljspeech\glow_tts\run-August-16-2024_06+21PM-204588f7

 > TRAINING (2024-08-16 18:22:04) 
Traceback (most recent call last):
  File "D:\AAAAA\myenv\lib\site-packages\trainer\trainer.py", line 1605, in fit
    self._fit()
  File "D:\AAAAA\myenv\lib\site-packages\trainer\trainer.py", line 1557, in _fit
    self.train_epoch()
  File "D:\AAAAA\myenv\lib\site-packages\trainer\trainer.py", line 1272, in train_epoch
    for cur_step, batch in enumerate(self.train_loader):
  File "D:\AAAAA\myenv\lib\site-packages\torch\utils\data\dataloader.py", line 440, in __iter__
    return self._get_iterator()
  File "D:\AAAAA\myenv\lib\site-packages\torch\utils\data\dataloader.py", line 388, in _get_iterator
    return _MultiProcessingDataLoaderIter(self)
  File "D:\AAAAA\myenv\lib\site-packages\torch\utils\data\dataloader.py", line 1038, in __init__
    w.start()
  File "D:\python\lib\multiprocessing\process.py", line 121, in start
    self._popen = self._Popen(self)
  File "D:\python\lib\multiprocessing\context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "D:\python\lib\multiprocessing\context.py", line 327, in _Popen
    return Popen(process_obj)
  File "D:\python\lib\multiprocessing\popen_spawn_win32.py", line 45, in __init__
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "D:\python\lib\multiprocessing\spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "D:\python\lib\multiprocessing\spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "D:\python\lib\multiprocessing\spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "D:\python\lib\multiprocessing\spawn.py", line 125, in _main
    prepare(preparation_data)
  File "D:\python\lib\multiprocessing\spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "D:\python\lib\multiprocessing\spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "D:\python\lib\runpy.py", line 268, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "D:\python\lib\runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "D:\python\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:\AAAAA\coqui-ai-TTS\recipes\ljspeech\glow_tts\train_glowtts.py", line 84, in <module>
    trainer.fit()
  File "D:\AAAAA\myenv\lib\site-packages\trainer\trainer.py", line 1632, in fit
    remove_experiment_folder(self.output_path)
  File "D:\AAAAA\myenv\lib\site-packages\trainer\generic_utils.py", line 78, in remove_experiment_folder
    fs.rm(experiment_path, recursive=True)
  File "D:\AAAAA\myenv\lib\site-packages\fsspec\implementations\local.py", line 179, in rm
    shutil.rmtree(p)
  File "D:\python\lib\shutil.py", line 757, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "D:\python\lib\shutil.py", line 627, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "D:\python\lib\shutil.py", line 625, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 32] 另一个程序正在使用此文件，进程无法访问。: 'D:/AAAAA/coqui-ai-TTS/recipes/ljspeech/glow_tts/run-August-16-2024_06+21PM-204588f7\\trainer_0_log.txt'
Hello, I'm encountering an error when trying to train the train_glowtts.py file using the LJ dataset. Could you please help me resolve this issue?

**Comment by anselmopro** [comment]
I am trying to install coqui TTS into Python "venv", but when installing "pip install TTS", got an error message: "ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (TTS)". Please help. Thank you! PS, I am using Windows 10, Python v3.11.1, and VS Code (updated version).

**Reply by eginhard** [reply]
Use [our fork](https://github.com/idiap/coqui-ai-TTS) (via `pip install coqui-tts`), it should install fine. This repo is not maintained anymore.

**Comment by anselmopro** [comment]
Thanks for your reply! I will try it. Best. PS, Happy New Year!

Yahoo Mail: Search, Organize, Conquer 
 
  On Mon, Dec 30, 2024 at 2:49 PM, Enno ***@***.***> wrote:   


Use our fork (via pip install coqui-tts), it should install fine. This repo is not maintained anymore.

**Comment by razaullah123** [comment]
I want to use TTS for English and Arabic text. But I don't find any suitable resource for these two. I am also looking for the free resources. I need help.

**Comment by Poupou2002** [comment]
Hi! Glad to be here. I’ve been testing different ways to improve TTS quality  lately using AudioModify to smooth out harsh or robotic tones. Works great alongside Coqui. Looking forward to sharing more!

**Comment by batvanio01** [comment]
Hello! I'm trying to retrain a model, but it won't accept the one I specified unless it's a checkpoint. Is there a place where I can download the model I need before it's been cleaned and optimized after training? I want to fine-tune: "tts_models/bg/cv/vits", but he ignores it and tries to do training from scratch. Thank you!

**Reply by eginhard** [reply]
You can use `--restore_path` to point to a specific checkpoint file from which to start fine-tuning: https://coqui-tts.readthedocs.io/en/latest/faq.html#difference-between-continue-path-and-restore-path

Note that the Vits model checkpoints only include the components needed for inference, so fine-tuning might not always work as expected.

**Comment by batvanio01** [comment]
I have a similar problem, but with a model trained in Russian with Bulgarian data because xtts it doesn't support my language. There is also a problem with some things being repeated twice and some pronouncing periods in a strange way. I deal with the periods by replacing them with an exclamation mark.

**Comment by KaushikNoukha** [comment]
Hi there everyone ive trained my xtts v2 finetuned model also tested it in the colab environment now ive hosted it in Huggingaface and tried ti use it in my local application i get errors like model warm up failed but the model is loading ,and failing due to warm up error C:\Users\admin\OneDrive\Desktop\Voice-model\intuitionX-voice-assistant\TTS\TTS\TTS
Successfully imported TTS from local repository (pip fallback disabled)
=== GET_TTS_SERVICE CALLED ===
=== CREATING NEW TTS SERVICE ===
=== TTS SERVICE INIT START ===
Initializing ULTRA-FAST Coqui TTS service...
Using device: cpu
[DIAG] TTS_LANGUAGE=en
[DIAG] TTS_SPEAKER_WAV resolved: C:\Users\admin\OneDrive\Desktop\Voice-model\intuitionX-voice-assistant\stt_server\Voice_samples_Gus_000_norm_fixed.wav
[TTS] Auto-downloading from Hugging Face: repo_id=kaushikaiml/Gus-finegrained rev=main files=(model.pth, config.json)
[TTS] Using HF-cached files:
  model: C:\hf_cache\hub\models--kaushikaiml--Gus-finegrained\snapshots\a2045b4c938865cbdf46886cb9341658599906da\model.pth
  config: C:\hf_cache\hub\models--kaushikaiml--Gus-finegrained\snapshots\a2045b4c938865cbdf46886cb9341658599906da\config.json
[TTS] Loading LOCAL checkpoint from environment paths...
[TTS] Effective model path: C:\hf_cache\hub\models--kaushikaiml--Gus-finegrained\snapshots\a2045b4c938865cbdf46886cb9341658599906da\model.pth
[TTS] Effective config path: C:\hf_cache\hub\models--kaushikaiml--Gus-finegrained\snapshots\a2045b4c938865cbdf46886cb9341658599906da\config.json
[TTS] Adjusted model path to directory for XTTS: C:\hf_cache\hub\models--kaushikaiml--Gus-finegrained\snapshots\a2045b4c938865cbdf46886cb9341658599906da
 > Using model: xtts
GPT2InferenceModel has generative capabilities, as `prepare_inputs_for_generation` is explicitly defined. However, it doesn't directly inherit from `GenerationMixin`. From 👉v4.50👈 onwards, `PreTrainedModel` will NOT inherit from `GenerationMixin`, and this model will lose the ability to call `generate` and other related functions.
  - If you're using `trust_remote_code=True`, you can get rid of this warning by loading the model with an auto class. See https://huggingface.co/docs/transformers/en/model_doc/auto#auto-classes
  - If you are the owner of the model architecture code, please modify your model class such that it inherits from `GenerationMixin` (after `PreTrainedModel`, otherwise you'll get an exception).
  - If you are not the owner of the model architecture class, please contact the model code owner to update it.
[DIAG] TTS python module path: C:\Users\admin\OneDrive\Desktop\Voice-model\intuitionX-voice-assistant\venv311\Lib\site-packages\TTS\__init__.py
[DIAG] Active TTS source: LOCAL checkpoint
[DIAG] self.tts.model_name:
Primary TTS model loaded: xtts_local
Extensive model warm-up for millisecond response...
 > Text splitted to sentences.
['Hello']
C:\Users\admin\OneDrive\Desktop\Voice-model\intuitionX-voice-assistant\venv311\Lib\site-packages\torchaudio\_backend\utils.py:213: UserWarning: In 2.9, this function's implementation will be changed to use torchaudio.load_with_torchcodec` under the hood. Some parameters like ``normalize``, ``format``, ``buffer_size``, and ``backend`` will be ignored. We recommend that you port your code to rely directly on TorchCodec's decoder instead: https://docs.pytorch.org/torchcodec/stable/generated/torchcodec.decoders.AudioDecoder.html#torchcodec.decoders.AudioDecoder.
  warnings.warn(
 > Text splitted to sentences.
['Hello']
 > Text splitted to sentences.
['Hello']
Warm-up failed for 'Hello': Model is multi-lingual but no `language` is provided.      
 > Text splitted to sentences.
['Hi there']
 > Text splitted to sentences.
['Hi there']
 > Text splitted to sentences.
['Hi there']
Warm-up failed for 'Hi there': Model is multi-lingual but no `language` is provided.  
 > Text splitted to sentences.
['Welcome']
 > Text splitted to sentences.
['Welcome']
 > Text splitted to sentences.
['Welcome']
Warm-up failed for 'Welcome': Model is multi-lingual but no `language` is provided.    
 > Text splitted to sentences.
['Thank you']
 > Text splitted to sentences.
['Thank you']
 > Text splitted to sentences.
['Thank you']
Warm-up failed for 'Thank you': Model is multi-lingual but no `language` is provided.  
 > Text splitted to sentences.
['Yes']
 > Text splitted to sentences.
['Yes']
 > Text splitted to sentences.
['Yes']
Warm-up failed for 'Yes': Model is multi-lingual but no `language` is provided.        
 > Text splitted to sentences.
['No']
 > Text splitted to sentences.
['No']
 > Text splitted to sentences.
['No']
Warm-up failed for 'No': Model is multi-lingual but no `language` is provided.
 > Text splitted to sentences.
['Okay']
 > Text splitted to sentences.
['Okay']
 > Text splitted to sentences.
['Okay']
Warm-up failed for 'Okay': Model is multi-lingual but no `language` is provided.      
 > Text splitted to sentences.
['Sure']
 > Text splitted to sentences.
['Sure']
 > Text splitted to sentences.
['Sure']
Warm-up failed for 'Sure': Model is multi-lingual but no `language` is provided.      
 > Text splitted to sentences.
['How are you?']
 > Text splitted to sentences.
['How are you?']
 > Text splitted to sentences.
['How are you?']
Warm-up failed for 'How are you?': Model is multi-lingual but no `language` is provided.
 > Text splitted to sentences.
["What's up?"]
 > Text splitted to sentences.
["What's up?"]
 > Text splitted to sentences.
["What's up?"]
Warm-up failed for 'What's up?': Model is multi-lingual but no `language` is provided.
 > Text splitted to sentences.
['Good morning']
 > Text splitted to sentences.
['Good morning']
 > Text splitted to sentences.
['Good morning']
Warm-up failed for 'Good morning': Model is multi-lingual but no `language` is provided.
 > Text splitted to sentences.
['Good evening']
 > Text splitted to sentences.
['Good evening']
 > Text splitted to sentences.
['Good evening']
Warm-up failed for 'Good evening': Model is multi-lingual but no `language` is provided.
 > Text splitted to sentences.
['I understand']
 > Text splitted to sentences.
['I understand']
 > Text splitted to sentences.
['I understand']
Warm-up failed for 'I understand': Model is multi-lingual but no `language` is provided.
 > Text splitted to sentences.
['Please wait']
 > Text splitted to sentences.
['Please wait']
 > Text splitted to sentences.
['Please wait']
Warm-up failed for 'Please wait': Model is multi-lingual but no `language` is provided.
 > Text splitted to sentences.
['Processing']
 > Text splitted to sentences.
['Processing']
 > Text splitted to sentences.
['Processing']
Warm-up failed for 'Processing': Model is multi-lingual but no `language` is provided.
 > Text splitted to sentences.
['Complete']
 > Text splitted to sentences.
['Complete']
 > Text splitted to sentences.
['Complete']
Warm-up failed for 'Complete': Model is multi-lingual but no `language` is provided.  
 > Text splitted to sentences.
['Error']
 > Text splitted to sentences.
['Error']
 > Text splitted to sentences.
['Error']
Warm-up failed for 'Error': Model is multi-lingual but no `language` is provided.      
 > Text splitted to sentences.
['Success']
 > Text splitted to sentences.
['Success']
 > Text splitted to sentences.
['Success']
Warm-up failed for 'Success': Model is multi-lingual but no `language` is provided.    
Model warm-up completed! 0 phrases cached.
=== TTS SERVICE CREATED, speaker_wav=C:\Users\admin\OneDrive\Desktop\Voice-model\intuitionX-voice-assistant\stt_server\Voice_samples_Gus_000_norm_fixed.wav ===
Generating speech for text: 'What about 800 culture?  Tell me more.'
Using model: xtts_local
[METRIC] Model source: LOCAL; Speaker: None
 > Text splitted to sentences.
['What about 800 culture?', 'Tell me more.']
[WARN] Direct tts_to_file failed: 'NoneType' object has no attribute 'encode'. Falling back to in-memory synthesis.
 > Text splitted to sentences.
['What about 800 culture?', 'Tell me more.']
 > Text splitted to sentences.
['What about 800 culture?', 'Tell me more.']
 > Text splitted to sentences.
['What about 800 culture?', 'Tell me more.']
[ERROR] In-memory synth fallback failed: Model is multi-lingual but no `language` is provided.
[METRIC] TTS synthesis time: 3185.0 ms
Error generating TTS: TTS engine failed to create output file: C:\Users\admin\OneDrive\Desktop\Voice-model\intuitionX-voice-assistant\audio_output\tts_output_20250915_134428.wav
[CHAT] TTS failed: Failed to generate TTS: TTS engine failed to create output file: C:\Users\admin\OneDrive\Desktop\Voice-model\intuitionX-voice-assistant\audio_output\tts_output_20250915_134428.wav
INFO:     127.0.0.1:56975 - "POST /chat HTTP/1.1" 200 OK   

Make sure to help me through this and this is my hf hosted model https://huggingface.co/kaushikaiml/Gus-finegrained

**Comment by keshavmishra-bit** [comment]
Is there any solution to handle phone numbers audio generation because i have seen that if there is repeated number in phone number then if you give space , comma or hyphen anything didn't work properly like i am facing skipping of digits sometimes and sometimes it speaks more than the digits are present there. So do you have a perfect solution for this problem?

**Comment by razaullah123** [comment]
Try to use word representation of the numbers. I mean give instruction to
the LLM to represent the numbers in word form i.e. 9 -> Nine, 8 -> eight.
This type of representation will solve the problem. Then TTS will speak it
correctly.

Thanks

**Reply by keshavmishra-bit** [reply]
No this converting to words didnt worked issue was still there i already tried that still some time it was repeating extra and some time skipping.
